extensions/amp-image-viewer/0.1/amp-image-viewer.js

```diff
@@ -183,66 +216,65 @@ export class ImageViewer {
    */
-  reset() {
-    this.image_.setAttribute('src', '');
-    Object.keys(this.ariaAttributes_).forEach(key => {
-      this.image_.removeAttribute(key);
-      this.ariaAttributes_[key] = null;
+  // TODO (cathyxz): test on mobile and verify this works.
+  registerOnResizeHandler_() {
+    const platform = Services.platformFor(this.win);
+
+    // Special case for iOS browsers due to Webkit bug #170595
+    // https://bugs.webkit.org/show_bug.cgi?id=170595
+    // Delay the onResize by 500 ms to ensure correct height and width
+    const debouncedOnResize = debounce(this.win, () => this.measure(), 500);
+
+    // Register an onResize handler to resize the image viewer
+    this.unlistenResize_ = this.getViewport().onResize(() => {
+      if (platform.isIos() && platform.isSafari()) {
+        debouncedOnResize();
+      } else {
+        this.measure();
+      }
     });
-    this.image_.removeAttribute('aria-describedby');
-    this.srcset_ = null;
-    this.imageBox_ = layoutRectLtwh(0, 0, 0, 0);
-    this.sourceWidth_ = 0;
-    this.sourceHeight_ = 0;
 
-    this.maxSeenScale_ = 1;
-    this.scale_ = 1;
-    this.startScale_ = 1;
-    this.maxScale_ = 2;
+    // iOS non-safari browsers do not reliably fire onResize on orientation
+    // change, so listen to orientationchange to trigger resize
+    if (platform.isIos() && !platform.isSafari()) {
+      this.unlistenOrientationChange_ = listen(this.win,
+          'orientationchange', debouncedOnResize);
+    }
+  }
 
-    this.startX_ = 0;
-    this.startY_ = 0;
-    this.posX_ = 0;
-    this.posY_ = 0;
-    this.minX_ = 0;
-    this.minY_ = 0;
-    this.maxX_ = 0;
-    this.maxY_ = 0;
+  /**
+   * @private
+   */
+  cleanupOnResizeHandler_() {
+    if (this.unlistenResize_) {
+      this.unlistenResize_();
+    }
 
-    if (this.motion_) {
-      this.motion_.halt();
+    if (this.unlistenOrientationChange_) {
+      this.unlistenOrientationChange_();
     }
-    this.motion_ = null;
   }
 
   /**
    * Initializes the image viewer to the target image element such as
    * "amp-img". The target image element may or may not yet have the img
    * element initialized.
-   * @param {!Element} sourceElement
-   * @param {?Element} sourceImage
    */
-  init(sourceElement, sourceImage = null) {
-    this.sourceWidth_ = sourceElement./*OK*/offsetWidth;
-    this.sourceHeight_ = sourceElement./*OK*/offsetHeight;
-    this.srcset_ = srcsetFromElement(sourceElement);
-
-    Object.keys(this.ariaAttributes_).forEach(key => {
-      this.ariaAttributes_[key] = sourceElement.getAttribute(key);
-      if (this.ariaAttributes_[key]) {
-        this.image_.setAttribute(key, this.ariaAttributes_[key]);
+  init_() {
+    this.sourceWidth_ = this.sourceElement_./*OK*/offsetWidth;
+    this.sourceHeight_ = this.sourceElement_./*OK*/offsetHeight;
+    this.srcset_ = srcsetFromElement(this.sourceElement_);
+
+    ARIA_ATTRIBUTES.forEach(key => {
+      if (this.sourceElement_.hasOwnProperty(key)) {
```

Comment:
Does this work? I'm not sure you can access an `Element`'s ARIA attributes using `hasOwnProperty`.